pkg/eval/builtin_special_test.go

```diff
@@ -147,6 +147,11 @@ func TestTmp(t *testing.T) {
 
 		That("var x; tmp x = y").DoesNotCompile(),
 		That("{ tmp x = y }").DoesNotCompile(),
+
+		That("{ tmp E:X = y; put $E:X }; has-env X").
+			Puts("y", false),
```

Comment:
AFAICT there is no test that expanding an undefined env var returns an empty string. This is a good time to add such a test and this test seems like a natural place to do it. Simple add `; put $E:X` to the end of the test and `""` as the third expected value.